hr_public_holidays/models/resource.py

```diff
@@ -0,0 +1,62 @@
+# ©  2018 Mihai Fekete <mihai.fekete@forbiom.eu>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import timedelta
+import pytz
+
+from odoo import api, fields, models
+
+
+def to_tz(datetime, tz_name):
+    tz = pytz.timezone(tz_name)
+    return pytz.UTC.localize(
+        datetime.replace(
+            tzinfo=None), is_dst=False).astimezone(tz).replace(tzinfo=None)
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    @api.multi
+    def _get_leave_intervals(self, resource_id=None,
+                             start_datetime=None, end_datetime=None):
+        """Add public holiday leaves, leaves without resource and calendar.
+
+        :return list leaves: list of time intervals """
+        res = super(ResourceCalendar, self)._get_leave_intervals(
+            resource_id, start_datetime, end_datetime)
+        self.ensure_one()
+        domain = [('resource_id', '=', False), ('calendar_id', '=', False)]
+        if start_datetime:
+            domain += [('date_to',
+                        '>',
+                        fields.Datetime.to_string(
+                            start_datetime + timedelta(days=-1)))]
+        if end_datetime:
+            domain += [('date_from',
+                        '<',
+                        fields.Datetime.to_string(
+                            start_datetime + timedelta(days=1)))]
+        leaves = self.env['resource.calendar.leaves'].search(domain)
+
+        filtered_leaves = self.env['resource.calendar.leaves']
+        for leave in leaves:
+            if start_datetime:
+                leave_date_to = to_tz(
+                    fields.Datetime.from_string(leave.date_to), leave.tz)
```

Comment:
If `leave.tz` is False, `to_tz` function will raise the error. Because user can create public holidays without timezone in `resource.calendar.leaves` record. I think we need to add a control for this.

Reply:
@hieulucky111 Code has been updated since your review comments and the requested change action, please review again